client/src/components/atoms/TagInput.tsx

```diff
@@ -0,0 +1,28 @@
+import React, { FunctionComponent, useState } from "react";
+import DeletableTag from "./DeletableTag";
+import { TextFieldWithAction } from "./TextFieldWithAction";
+
+interface TagInputProps {
+    tagStrings: string[],
+    onChange: (value: string) => boolean,
+    onSubmit: (value: string) => void,
+    onDelete: (id: number) => void,
+    isErroneous: boolean
+}
+
+const TagInput: FunctionComponent<TagInputProps> = (props: TagInputProps) => {
+    return (
+        <div className="tag-input">
+            <div className="text-field-action-container">            
+                <TextFieldWithAction isErroneous={(!props.isErroneous)} onSubmit={(value:string) => props.onSubmit(value)} onChange={(value: string)=> props.onChange(value)} showAction={props.isErroneous} placeholder="Enter a new type" type="text" actionString="Add new type:" iconClassName="bi bi-arrow-return-left"></TextFieldWithAction>
```

Comment:
We should also avoid showing the action if the input in the text field is empty.
I suggest that we keep track of the input in the components stats
```
const [input, setInput] = useState("")

<TextFieldWithAction 
 onSubmit={(value:string) => {setInput(value); setSubmitprops.onSubmit(value)}} 
onChange={(value: string)=> {setInput(value); props.onChange(value)}} 
showAction={props.isErroneous || input.length > 0 } 
></TextFieldWithAction>
```